src/app/ReadHandler.cpp

```diff
@@ -421,6 +431,39 @@ CHIP_ERROR ReadHandler::ProcessAttributePathList(AttributePathIBs::Parser & aAtt
     return err;
 }
 
+CHIP_ERROR ReadHandler::ProcessDataVersionFilterList(DataVersionFilterIBs::Parser & aDataVersionFilterListParser)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    TLV::TLVReader reader;
+    aDataVersionFilterListParser.GetReader(&reader);
+
+    while (CHIP_NO_ERROR == (err = reader.Next()))
+    {
+        VerifyOrReturnError(TLV::AnonymousTag() == reader.GetTag(), CHIP_ERROR_INVALID_TLV_TAG);
+        ClusterInfo clusterInfo;
+        ClusterPathIB::Parser path;
+        DataVersionFilterIB::Parser filter;
+        ReturnErrorOnFailure(filter.Init(reader));
+        ReturnErrorOnFailure(filter.GetDataVersion(&(clusterInfo.mDataVersion)));
+        ReturnErrorOnFailure(filter.GetPath(&path));
+        ReturnErrorOnFailure(path.GetEndpoint(&(clusterInfo.mEndpointId)));
+        ReturnErrorOnFailure(path.GetCluster(&(clusterInfo.mClusterId)));
+        VerifyOrReturnError(clusterInfo.IsValidDataVersionFilter(), CHIP_ERROR_IM_MALFORMED_DATA_VERSION_FILTER_IB);
+
+        err = InteractionModelEngine::GetInstance()->PushFront(mpAttributeClusterInfoList, clusterInfo);
```

Comment:
This is ... not the right list?

Was this tested at all?  Because I don't see how it could possibly have worked.

Reply:
missed the test for this logic....fixed it, thanks